src/main/java/org/dependencytrack/notification/vo/ProjectVulnerabilityUpdate.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * This file is part of Dependency-Track.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.dependencytrack.notification.vo;
+
+import org.dependencytrack.model.Component;
+import org.dependencytrack.model.Vulnerability;
+import org.dependencytrack.model.VulnerabilityUpdateDiff;
+
+public class ProjectVulnerabilityUpdate {
```

Comment:
Consider making this a [`record`](https://docs.oracle.com/en/java/javase/21/language/records.html) instead.

I'd further recommend to rename this class to `VulnerabilityUpdated`, since it's not specific to a project, as the current name would suggest.

Reply:
By using `ProjectVulnerabilityUpdate` instead of `VulnerabilityUpdate` I was trying to make it clear that the notification only represents an update to a vulnerability that affects a project in a portfolio. I thought it might be confusing to call it `VulnerabilityUpdate`, as that might lead someone to think that the notification would fire on any updates to a vulnerability in the vuln DB. Perhaps that's already clear enough though from the notification being scoped to the `PORTFOLIO`?

Reply:
Ah, I see, that reasoning makes sense. IMO, `VulnerabilityUpdated` would still be less ambiguous than `ProjectVulnerabilityUpdated`. I guess there are still better ways to word the intention though, but my non-native-speaker skills are not helping.

I was thinking something like these, although they still sound a bit awkward:

* `AffectingVulnerabilityUpdated` - i.e. a vuln *affecting* something was updated
* `FindingVulnerabilityUpdated` - i.e. a vuln that is part of a finding was updated

Reply:
Out of those two suggestions I think `FindingVulnerabilityUpdated` is better, but still might be confusing as someone might expect to receive a `Finding` in the event payload (which has a specific meaning and schema as an API resource).

Is `ActiveVulnerabilityUpdated` any better? It mirrors the idea of 'Active' and 'Inactive' projects.